jsapp/js/account/organization/membersQuery.ts

```diff
@@ -34,6 +34,34 @@ export interface OrganizationMember {
   };
 }
 
+/**
+ * For updating member within given organization. Accepts partial properties
+ * of `OrganizationMember`.
+ */
+export async function patchOrganizationMember(
+  organizationId: string,
+  username: string,
+  newMemberData: Partial<OrganizationMember>
+) {
+  const apiUrl = endpoints.ORGANIZATION_MEMBER_URL
+    .replace(':organization_id', organizationId)
+    .replace(':username', username);
+  return fetchPatch<OrganizationMember>(apiUrl, newMemberData);
+}
+
+/**
+ * For removing member from given organization.
+ */
+export async function removeOrganizationMember(
+  organizationId: string,
+  username: string
+) {
+  const apiUrl = endpoints.ORGANIZATION_MEMBER_URL
+    .replace(':organization_id', organizationId)
+    .replace(':username', username);
+  return fetchDelete(apiUrl);
+}
+
```

Comment:
We will probably need to add to these file the mutation hooks for the functions  and do not export the actual functions above.

``` typescript

/**
 * Mutation hook to patch and organization member.
 * This receives the current members query result to refetch the list after the
 * mutation is successful instead of using query keys duo to pagination parameters.
 */
export const usePatchOrganizationMember = (
  membersQuery: UseQueryResult<PaginatedResponse<OrganizationMember>>
) => {
  const orgQuery = useOrganizationQuery();
  const orgId = orgQuery.data?.id;

  return useMutation({
    mutationFn: ({
      username,
      data,
    }: {
      username: string;
      data: Partial<OrganizationMember>;
    }) => {
      if (!orgId) {
        throw new Error('Organization not found');
      }
      return patchOrganizationMember(orgId, username, data);
    },
    onSuccess: () => {
      // This will refetch the members list
      membersQuery.refetch();
    },
  });
};

/**
 * Mutation hook to remove an organization member.
 * This receives the current members query result to refetch the list after the
 * mutation is successful instead of using query keys duo to pagination parameters.
 */
export const useRemoveOrganizationMember = (
  membersQuery: UseQueryResult<PaginatedResponse<OrganizationMember>>
) => {
  const orgQuery = useOrganizationQuery();
  const orgId = orgQuery.data?.id;

  return useMutation({
    mutationFn: (username: string) => {
      if (!orgId) {
        throw new Error('Organization not found');
      }
      return removeOrganizationMember(orgId, username);
    },
    onSuccess: () => {
      // This will refetch the members list
      membersQuery.refetch();
    },
  });
};

```

Usage would be something like: 
``` typescript
const membersQuery = useOrganizationMembersQuery()
const membersPatch = usePatchOrganizationMember(membersQuery)

const onPatch = (username: string, data:Partial<OrganizationMember>) => {
    membersPatch.mutateAsync({username, data})
}
```

Reply:
@pauloamorimbr Thansk for your guidance on this! I did it a bit differently though - not requiring the `membersQuery` to be passed, and using invalidation rather than refetch (see comment link)

Reply:
@Akuukis wouldn't this be the options we want?:

```ts
/**
 * Mutation hook for updating organization member. It ensures that all related
 * queries refetch data (are invalidated).
 */
export function usePatchOrganizationMember(options?: Omit<UseMutationOptions<unknown, Error, PatchOrgMemberVars>, 'mutationFn' | 'onSettled'>) {
  const queryClient = useQueryClient();

  return useMutation({
    mutationFn: (vars: PatchOrgMemberVars) => patchOrganizationMember(vars),
    onSettled: () => {
      // We invalidate query, so it will refetch (instead of refetching it
      // directly, see: https://github.com/TanStack/query/discussions/2468)
      queryClient.invalidateQueries({queryKey: [QueryKeys.organizationMembers]});
    },
    ...options,
  });
}
```

Not arguing, just verifying if I understand correctly :) Changes:
- I removed the 4th `unknown` as I don't know why it's there.
- I changed the 1st to `unknown` as I don't use the mutation fn output in any way.
- I changed `'mutationFn'` to `'mutationFn' | 'onSettled'`, as both are predefined.

And general question - do we even need to have `options` here given the fact we are not using them? It adds some "noise" to the code. I don't have an opinion on this, just asking

Reply:
```ts
/**
 * Mutation hook for updating organization member. It ensures that all related
 * queries refetch data (are invalidated).
 */
export function usePatchOrganizationMember(defaultVars?: Partial<PatchOrgMemberVars>) {
  const queryClient = useQueryClient();

  return useMutation({
    mutationFn: (vars: PatchOrgMemberVars) => patchOrganizationMember({...defaultVars, vars}),
    onSettled: () => {
      // We invalidate query, so it will refetch (instead of refetching it
      // directly, see: https://github.com/TanStack/query/discussions/2468)
      queryClient.invalidateQueries({queryKey: [QueryKeys.organizationMembers]});
    },
    ...options,
  });
}
```

Reply:
```ts
export default (props: {orgId}) => {
  const patchOrganizationMember = usePatchOrganizationMember({orgId})
  const handleCSubmit = useCallback((event) => patchOrganizationMember({name: event.target.value}), [])
  return <input onSubmit={handleCSubmit} />
  }